### ISSUE TITLE
Restore hot-exit files to the correct instance when renaming workspace folder

### ISSUE THREAD

**Issue by Binly42** [Microsoft/vscode#26533]
- VSCode Version: 1.12.1
- OS Version: Windows7 64bit SP1 with latest update

Steps to Reproduce:

1. open a folder, and then open and modify some files ;
2. (under the setting {"files.hotExit": "onExitAndWindowClose"} ,) close the window ;
3. rename the folder (, or change its location) ;
4. open that folder with vscode ;
5. the restoring fails ;

*6. if undo the rename and/or move of the folder, everything seems to be fine again ;

So, imho maybe the archive about hotExit (or some related params (likethe current path of the folder to be restored) ) should be stored in the ./.vscode ?

**Comment by Emeryao** [comment]
I think your request is little *MEAN* for VS Code

**Comment by bpasero** [comment]
@Tyriar we should be able to restore the files maybe as untitled files to one window so that at least the data is not lost?

**Comment by slodki** [comment]
look at #26545

**Comment by bpasero** [comment]
Ok this is not data loss but inconvenience: we do open the dirty files in a separate window as untitled files and not within the instance where the renamed folder was opened. I do not think we could actually support this in any way unless we would track file events between restarts of VS Code.

**Comment by Binly42** [comment]
Actually I can understand that is not data loss. If it did lose data, I would not be able to see the 6-th step among the steps-to-reproduce in my first post.
So I guess it is just a error that cannot find any saved archive when addressing to the changed new path.

If my understand is not wrong, it may be not necessary to track the file events (on VSCode's own). 
What I imagine is that VSCode records and restores the status of the folder relying on the path. Instead, working with a certain identifier, which is independent of the physical path, for example a unique id established whenever the folder was opened at the first time. I think this may work ...

Also I know this may be a really little need/feature for what the name *hot-exit* suggests ...

I am sorry for my poor english ...

**Comment by Tyriar** [comment]
@bpasero we were talking about a command to open workspaces that have backups, I still believe that would be useful and could help with this.

As for tracking the change in the folder I can't think of a nice way to do it. We could mitigate the problem by providing a command like above and maybe a command that opened the actual backup folder in explorer/finder so users don't need to hunt for them manually.

**Comment by ctsstc** [comment]
I'm not sure if this is related, it seems to be semi-related; I can open a new issue if needs be.

I created a new window that did not yet have a directory opened (there was another window opened with a project/directory), I created a new file in this new window, the file was populated but not saved, my system restarted for updates and I lost the file. So is that file saved somewhere to some odd namespace and I can restore it, or is it gone forever?
